### DISCUSSION TITLE
Create re-usable component

### DISCUSSION THREAD

**Discussion by gmwill934** [adobe/react-spectrum#3717] [Q&A]
Hello

I'm trying to create a re-usable date picker component with typescript, no documentation could I find on the docs, I only saw JS examples.

I'm getting this error when trying to use the `AriaDatePickerProps` as a prop to a react component.

```ts
export function DateField<T>(props: AriaDatePickerProps<T>) {
   ...
}
```

It looks like I need to extend the `T` generic with some other types that I cannot find on `react-aria`.

Is there a simple approach to this?

**Comment by snowystinger** [comment]
See https://reactspectrum.blob.core.windows.net/reactspectrum/5d768a0d443a59cf84e67e2297665ca80bc121b9/docs/react-aria/useDatePicker.html

From this PR where we are implementing TS examples https://github.com/adobe/react-spectrum/pull/3710
Let us know if that addresses your issue

**Reply by devongovett** [reply]
(Those examples don't have TS yet @snowystinger 😉)

**Reply by snowystinger** [reply]
Ah, yes, but I think what you're looking for is in the API definition at the top too
